File: src/main/java/com/oshi/libgenericsettings/viewholder/SettingsTitleSeekbarMinMaxViewHolder.java
package com.oshi.libgenericsettings.viewholder;

import android.databinding.ViewDataBinding;
import android.support.v7.widget.AppCompatSeekBar;
import android.widget.SeekBar;

import com.oshi.libgenericsettings.BR;
import com.oshi.libgenericsettings.R;
import com.oshi.libgenericsettings.data.TitleSeekBarMinMaxData;
import com.oshi.libgenericsettings.presenter.ISettingsPresenter;

public class SettingsTitleSeekbarMinMaxViewHolder extends BaseSettingsViewHolder<TitleSeekBarMinMaxData> {

    private AppCompatSeekBar appCompatSeekBar;

    public SettingsTitleSeekbarMinMaxViewHolder(ViewDataBinding viewDataBinding) {
        super(viewDataBinding);

        appCompatSeekBar = viewDataBinding.getRoot().findViewById(R.id.seekBar);
    }

    @Override
    public void onBind(final TitleSeekBarMinMaxData data, final ISettingsPresenter presenter, final int position) {
        viewDataBinding.setVariable(BR.titleSeekBarMinMaxData, data);
        viewDataBinding.setVariable(BR.settingsPresenter, presenter);
        viewDataBinding.setVariable(BR.itemPosition, position);
        viewDataBinding.executePendingBindings();

        appCompatSeekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                if (fromUser) {
                    data.setInternalProgress(progress);
                    presenter.onTitleSeekBarMinMaxChanged(viewDataBinding.getRoot(), data, position);
                    // Do not execute pending binding as progress is need to be initiated only once
                }
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {

            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {

            }
        });


    }
}

File: src/main/java/com/oshi/libgenericsettings/viewholder/SettingsTitleSeekbarMinMaxViewHolder.kt
package com.oshi.libgenericsettings.viewholder

import android.databinding.ViewDataBinding
import android.support.v7.widget.AppCompatSeekBar
import android.widget.SeekBar
import com.oshi.libgenericsettings.BR
import com.oshi.libgenericsettings.R
import com.oshi.libgenericsettings.data.TitleSeekBarMinMaxData
import com.oshi.libgenericsettings.databinding.ViewTypeTitleSeekbarMinMaxBinding
import com.oshi.libgenericsettings.presenter.ISettingsPresenter

class SettingsTitleSeekbarMinMaxViewHolder(viewDataBinding: ViewDataBinding) : BaseSettingsViewHolder<TitleSeekBarMinMaxData>(viewDataBinding), SeekBar.OnSeekBarChangeListener {

    private var binding = viewDataBinding as ViewTypeTitleSeekbarMinMaxBinding

    private lateinit var recycledData : TitleSeekBarMinMaxData
    private lateinit var recycledPresenter: ISettingsPresenter

    init {
        binding.seekBar.setOnSeekBarChangeListener(this)
    }

    override fun onBind(data: TitleSeekBarMinMaxData, presenter: ISettingsPresenter, position: Int) {
        recycledData = data
        recycledPresenter = presenter

        viewDataBinding.setVariable(BR.titleSeekBarMinMaxData, data)
        viewDataBinding.setVariable(BR.settingsPresenter, presenter)
        viewDataBinding.setVariable(BR.itemPosition, position)
        viewDataBinding.executePendingBindings()

    }

    override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
        if (fromUser) {
            recycledData.setInternalProgress(progress)
            recycledPresenter.onTitleSeekBarMinMaxChanged(viewDataBinding.root, recycledData, adapterPosition)
            // Do not execute pending binding as progress is need to be initiated only once
        }
    }

    override fun onStartTrackingTouch(seekBar: SeekBar?) {}

    override fun onStopTrackingTouch(seekBar: SeekBar?) {}
}